pkg/health/health.go

```diff
@@ -50,26 +52,34 @@ type healthCheckOptions struct {
 
 // StartEndpointHealthCheck starts a health check on the specified address with the given options.
 // It returns a channel that will emit true if the endpoint is healthy and false if the failure conditions
-// Have been met.
-func StartEndpointHealthCheck(ctx context.Context, endpointAddress string, opts ...Option) <-chan bool {
+// have been met.
+func StartEndpointHealthCheck(ctx context.Context, opts ...Option) <-chan bool {
 	options := &healthCheckOptions{}
 	applyDefaults(options)
-
 	for _, o := range opts {
 		o(options)
 	}
+
+	if options.address == "" {
+		panic("required option 'address' is missing")
+	}
```

Comment:
Never panic, return error instead.

```suggestion
	if options.address == "" {
		return nil, errors.New("required option 'address' is missing")
	}
```

Reply:
If this panics, it's a development-time error, so it is appropriate to panic here IMHO.

Reply:
Then is there a need to panic at all? It will error later anyway.

Reply:
There's no _need_ but again this is just a sanity-check for development-time bugs